builtin/providers/aws/resource_aws_lambda_permission.go

```diff
@@ -182,7 +182,14 @@ func resourceAwsLambdaPermissionRead(d *schema.ResourceData, meta interface{}) e
 		statement, err = findLambdaPolicyStatementById(&policy, d.Id())
 		return err
 	})
+
 	if err != nil {
+		if strings.HasPrefix(err.Error(), "timeout while waiting for state") {
```

Comment:
Yeah, the cleaner way would be probably creating a special `error` type instead of using `fmt.Errorf` here: https://github.com/hashicorp/terraform/blob/master/helper/resource/state.go#L135-L137

then we could be comparing something like `err.Code()` instead.